### PR TITLE
Fix pullSecrets, fix assignment of NET_BIND_SERVICE capability

### PR DESCRIPTION
**Pull request by firefrei**
I tried to use the helm chart in v1.19.7 on our RKE2-based Kubernetes cluster which has special hardening according to CIS-1.6 implemented. However, the images could not be pulled from our private repo as well as the pods terminated with "operation not permitted".

The reason for these two errors were:
- imagePullSecrets were not used (mentioned in values.yaml as image.pullSecrets, but not evaluated in template)
- the NET_BIND_SERVICE capability was not set in securityContext of pod
  - securityContext was not mentioned in values.yaml, however it was evaluated in template.
- in the PSP element, the capability was named incorrectly. It was CAP_NET_BIND_SERVICE, however, due to official K8s docs, there should be no `CAP_` prefix.

This MR fixes all these issues. It tested the helm chart with these fixes, the pods are now able to start!

### CODE REVIEW

**Comment by firefrei** [comment]
Hmmm... the linter complains that I did not bump the version number. Should I do this?

```
 ✖︎ coredns => (version: "1.19.7", path: "charts/coredns") > chart version not ok. Needs a version bump! 
```

**Comment by firefrei** [comment]
@mrueg Okay, so when I have `serviceAccount.create` set to false, which is the default, then I cannot use an imagePullSecret? What's the idea behind this? 

https://github.com/coredns/helm/blob/2fd88c262f883adee5978f8b8923f72a04e2264b/charts/coredns/templates/serviceaccount.yaml#L1

**Comment by firefrei** [comment]
@mrueg how should we proceed with this PR?
What is the desired way of using imagePullSecrets? I would suggest to use it with the Deployment, to not rely on the service account.
Should I extend the PR and remove it from the ServiceAccount? Should I remove the imagePullSecret again from the Deployment and limit this PR to the NET_BIND_SERVIC fix?

**Comment by HagaiBarel** [comment]
Hi @firefrei I agree with you that specifying a `pullSecret` is much more common on the deployment than on the service account, because the service account is conditional. I think we can proceed with the PR as-is. Please rebase and bump the chart version by a minor version

**Comment by hagaibarel** [comment]
Thanks, looking good, one last thing is to update the changelog, see https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#changelog for instructions on what and how

**Comment by firefrei** [comment]
Done! I hope I inserted the item at the correct position in in the list :-)

File: charts/coredns/Chart.yaml [hunk 1]
```diff
@@ -21,3 +21,5 @@ annotations:
   artifacthub.io/changes: |
     - kind: fixed
       description: Fix HPA configuration
+    - kind: fixed
+      description: Fix assignment of NET_BIND_SERVICE capability, enable use of pullSecrets with Deployment
```
Comment (hagaibarel) [inline charts/coredns/Chart.yaml:25]: This should completely replace the current change log, not append to it. Also, please break it down to 2 items. The final structure should look similar to:

```yaml
artifacthub.io/changes: |
  - kind: fixed
    description: Fix assignment of NET_BIND_SERVICE capability
  - kind: fixed
    description: Enable use of pullSecrets with Deployment
```

File: charts/coredns/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: coredns
-version: 1.20.2
+version: 1.21.0
```
Comment (hagaibarel) [inline charts/coredns/Chart.yaml:3]: We'll need another version increase based on the most recent chart version

**Review by hagaibarel — CHANGES_REQUESTED** [review]

**Comment by firefrei** [comment]
Sorry for the delay,
next try :-)

**Review by hagaibarel — APPROVED** [review]

**Pull request merged**